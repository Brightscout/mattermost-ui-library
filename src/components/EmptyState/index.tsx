import React from 'react';

type EmptyStatePropTypes = {
    title: string;
    subTitle?: string;
    buttonConfig?:{
        text: string,
        link?: string,
        download?: boolean,
        action?: (event: React.MouseEvent<HTMLButtonElement>) => void,
    } | null;
    iconClass?: string;
    icon?: JSX.Element;
    className?: string;
}

const EmptyState = ({
    title,
    subTitle,
    buttonConfig,
    iconClass,
    icon,
    className,
}: EmptyStatePropTypes) => (
    <div className={`empty-state text-center d-flex align-items-center justify-content-center ${className}`}>
        <div className='d-flex flex-column align-items-center'>
            <div className='empty-state__icon margin-bottom-25 d-flex justify-content-center align-items-center'>
                {icon ?? <i className={iconClass ?? 'fa fa-wifi'}/>}
            </div>
            <p className='empty-state__title wt-500 font-size-20'>{title}</p>
            {subTitle && <p className='empty-state__subtitle margin-top-15'>{subTitle}</p>}
            {buttonConfig?.action && (
                <button
                    onClick={buttonConfig.action}
                    className='empty-state__btn btn btn-primary margin-top-25'
                >
                    {buttonConfig.text}
                </button>
            )}
            {buttonConfig?.link && (
                <a
                    target='_blank'
                    rel='noreferrer'
                    href={buttonConfig.link}
                    download={buttonConfig.download}
                    className='empty-state__btn btn btn-primary margin-top-25'
                >
                    {buttonConfig.text}
                </a>
            )}
        </div>
    </div>
);

export default EmptyState;
