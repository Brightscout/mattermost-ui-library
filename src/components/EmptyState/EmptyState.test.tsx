import React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';

import EmptyState from '.';

// Mock props data
const buttonActionHandler = jest.fn();
const emptyStateProps = {
    title: 'Empty state title',
    subTitle: 'Empty state subtitle',
    buttonConfig: {
        text: 'Button Text',
        link: 'button link',
        download: true,
        action: buttonActionHandler,
    },
    icon: <i className='fa fa-link'/>,
    className: 'empty-state-class',
};

describe('EmptyState', () => {
    let component: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

    beforeEach(() => {
        component = shallow(<EmptyState {...emptyStateProps}/>);
    });

    it('Should render correctly', () => {
        expect(component).toMatchSnapshot();
    });

    it('Should apply the passed className prop', () => {
        expect(component.hasClass(emptyStateProps.className)).toBeTruthy();
    });

    it('Should render the icon correctly', () => {
        expect(component.contains(emptyStateProps.icon)).toBeTruthy();
    });

    it('Should render the title correctly', () => {
        expect(component.text().includes(emptyStateProps.title)).toBeTruthy();
    });

    it('Should render the subTitle correctly', () => {
        expect(component.text().includes(emptyStateProps.subTitle)).toBeTruthy();
    });

    it('Should render the button text correctly', () => {
        expect(component.find('button.empty-state__btn').text().includes(emptyStateProps.buttonConfig.text)).toBeTruthy();
    });

    it('Should call the click handler when the button is clicked', () => {
        component.find('button.empty-state__btn').simulate('click');
        expect(buttonActionHandler).toHaveBeenCalledTimes(1);
    });

    it('Should render the link correctly', () => {
        expect(component.find('a.empty-state__btn').text().includes(emptyStateProps.buttonConfig.text)).toBeTruthy();
    });
});
