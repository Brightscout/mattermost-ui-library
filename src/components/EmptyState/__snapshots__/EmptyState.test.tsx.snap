// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState Should render correctly 1`] = `
<div
  className="empty-state text-center d-flex align-items-center justify-content-center empty-state-class"
>
  <div
    className="d-flex flex-column align-items-center"
  >
    <div
      className="empty-state__icon margin-bottom-25 d-flex justify-content-center align-items-center"
    >
      <i
        className="fa fa-link"
      />
    </div>
    <p
      className="empty-state__title wt-500 font-size-20"
    >
      Empty state title
    </p>
    <p
      className="empty-state__subtitle margin-top-15"
    >
      Empty state subtitle
    </p>
    <button
      className="empty-state__btn btn btn-primary margin-top-25"
      onClick={[MockFunction]}
    >
      Button Text
    </button>
    <a
      className="empty-state__btn btn btn-primary margin-top-25"
      download={true}
      href="button link"
      rel="noreferrer"
      target="_blank"
    >
      Button Text
    </a>
  </div>
</div>
`;
